kernel/src/table_changes/scan.rs

```diff
@@ -153,9 +164,27 @@ impl TableChangesScanBuilder {
             predicate: self.predicate,
             all_fields,
             have_partition_cols,
+            physical_schema: StructType::new(read_fields).into(),
         })
     }
 }
+
+impl TableChangesScan {
+    #[allow(unused)]
+    pub(crate) fn scan_data(
```

Comment:
nit: docs (can we say that it gives an iterator which yields at each commit boundary and for each commit is producing a batch of data where each row is possibly filtered but includes the actions ultimately needed to read CDF and some additional metadata)